server/routerlicious/packages/services/src/mongoExceptionRetryRules/mongoError/index.ts

```diff
@@ -298,6 +311,42 @@ class ConnectionClosedMongoErrorRule extends BaseMongoExceptionRetryRule {
 	}
 }
 
+class ConnectionTimedOutBulkWriteErrorRule extends BaseMongoExceptionRetryRule {
+	private static readonly errorName = "MongoBulkWriteError";
```

Comment:
I think we should not retry on this error, and instead, at caller level, handle the retry logic by its own. (I know scriptorium and objectstore are the consumers)

The reason is if we write [1,10] into db, and [1,4] and [6,10] success, this retry logic would rewrite everything again and got id already exist error.

`insertMany` has more specific logic, so we should:
1. not retry on this error, and bubble up to have caller to handle.
2. Or retry here, but in facade level, provide general retry logic, that is filter out the payload and only keep these failed only.

Reply:
That makes sense and I agree. This PR is just to avoid restarting on time out errors similar to "RequestTimedOutBulkWriteErrorRule" for previous sdk. 
I agree the retry logic should filter out the already inserted ids for this error - let me see if we can include it with this change or if it requires more changes then will do that in a separate PR.

Reply:
Discussed offline, will create a separate PR for this change.